service/raft/group0_upgrade.hh

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+namespace service {
+
+enum class group0_upgrade_state : uint8_t {
```

Comment:
This is the best place to document these states, their purpose and relationship.